src/interfaces.ts

```diff
@@ -24,10 +29,24 @@ export interface BatchUpdateDocumentsOptions {
    *
    * Make sure your API key is granted the privilege to EDIT fields before using this option.
    * See https://docs.coveo.com/en/1707#fields-domain
+   *
+   * @default true
    */
   createFields?: boolean;
 }
 
+export interface ParseDocumentOptions {
+  /**
+   * The {@link Transformer} to apply to the fields found in the parsed documents.
+   * If not specified, no transformation will be applied to the field names.
+   * A Transformer is useful to prevent getting errors whenever some of field names contain special characters or any other element not supported by the Platform.
+   *
+   * For a list of built-in transformers, use {@link BuiltInTransformers}.
+   *
+   * @default BuiltInTransformers.identity
+   */
+  fieldNameTransformer?: Transformer;
+}
 export interface ConcurrentProcessing {
   /**
    * The maximum number of requests to send concurrently to the Coveo platform.
```

Comment:
While at it, let's add `@default 10 and remove the verbose 'the default value is set to'